internal/mssql/sqlclient.go

```diff
@@ -277,3 +278,105 @@ func (m client) UnassignRole(ctx context.Context, role string, principal string)
 
 	return err
 }
+
+func (m client) ReadDatabasePermission(ctx context.Context, id string) (DatabaseGrantPermission, error) {
+	var DatabaseGrantPermission DatabaseGrantPermission
+	principal := strings.Split(id, "/")[0]
+	permission := strings.Split(id, "/")[1]
+
+	cmd := `
+		SELECT
+			dp.[name] AS [principal],
+			sdp.[permission_name] AS [permission]
+		FROM
+			sys.database_permissions AS sdp
+		JOIN
+			sys.database_principals AS dp ON sdp.grantee_principal_id = dp.principal_id
+		WHERE
+			sdp.[class] = 0
+			AND sdp.[state] IN ('G', 'W')
+			AND dp.[name] = @p1
+			AND sdp.[permission_name] = @p2;
+	`
+
+	tflog.Debug(ctx, fmt.Sprintf("Reading DB permission [principal: %s, permission: %s]", principal, permission))
+	result := m.conn.QueryRowContext(ctx, cmd, principal, permission)
+
+	err := result.Scan(&DatabaseGrantPermission.Principal, &DatabaseGrantPermission.Permission)
+	if err != nil {
+		tflog.Warn(ctx, fmt.Sprintf("failed to scan result: %v", err))
+		return DatabaseGrantPermission, err
+	}
+
+	DatabaseGrantPermission.Id = fmt.Sprintf("%s/%s", DatabaseGrantPermission.Principal, strings.ToLower(DatabaseGrantPermission.Permission))
+	tflog.Debug(ctx, fmt.Sprintf("SUCCESS Reading DB permission principal: %s, permission: %s", principal, permission))
+
+	return DatabaseGrantPermission, nil
+}
+
+func (m client) GrantDatabasePermission(ctx context.Context, principal string, permission string) (DatabaseGrantPermission, error) {
+	var DatabaseGrantPermission DatabaseGrantPermission
+
+	query := fmt.Sprintf("GRANT %s TO %s", permission, principal)
```

Comment:
This needs to follow the pattern for CREATE USER to generate the statement, including QUOTENAME and other things.

Otherwise this can result in SQL Injection issues. It's rare because this is taking input from Terraform, not random users, but it should be correct to eliminate any possibility.